### ISSUE TITLE
Improve UX for password validation

### ISSUE THREAD

**Issue by robinv85** [FightPandemics/FightPandemics#395]
There could be 2 additional validations for password strength, because currently only the character length is checked (I think client-side only at the moment), so for example:
- add client-side password strength validation: checking at least 1 special character / capitalized letter
- add API-level validation in the JSON schema

Both could perhaps use the same regex if possible
This should match exactly the same validation as the one on Auth0 (which can be configured as well)

**Comment by joshmorel** [comment]
@robinv85 this is implemented F/E now (https://github.com/FightPandemics/FightPandemics/blob/1ae090dc0a094b979c1cc579aab194e45b9a6f56/client/src/utils/validators.js#L12) but not on API-level validation (although also checked by Auth0). 

Should we close this one as resolved and open a low priority one for B/E or are we good?

**Comment by robinv85** [comment]
@joshmorel yeah so I remember adding that more detailed password validation for client-side validation in the forms. 

I thought it was also asked by designers to show the validation rules below the text input when signing up, and not only show the error message if it's invalid. (so that is also "improving UX" for the password validation).

There are also mobile apps that should get the same validation but if the backend doesn't have validation in the API schema then there is still Auth0 who has the final call about this so I guess it's rather low priority indeed.

Not sure why it would be better to close this and create a new ticket though.

**Comment by joshmorel** [comment]
@robinv85 My rationale was to have a backend specific issue since the F/E seems to be addressed.

**Comment by robinv85** [comment]
@joshmorel fine by me. We could also update thi issues and specify that it's for the backend only.
The other thing that I mentioned, not sure if that is fixed: "show the password validation rules to the user, before they have to enter it", so probably as a green text below the input field.
Since that was mentioned on Slack but might be be clearly shown in the designs or in the requirements.
